File: classTypes4/c#/Gato.cs
public class Gato : CarnivoroTerrestre, IDomestico{
    private string raza;
    
    // constructor 1: inicializa con parametros basicos
    public Gato(string nombreCientifico, double pesoKg, string raza) 
        : base(nombreCientifico, pesoKg, 2){
        this.raza = raza;
    }
    
    // constructor 2: sobrecarga, inicializa solo con el peso y usa valores por defecto para otros
    public Gato(double pesoKg) 
        // c# usa this para llamar a un constructor de la misma clase
        : this("felis catus", pesoKg, "comun"){

        Console.WriteLine("sobrecarga usada: gato comun creado con peso especifico.");
    }

    // metodos obligatorios implementados, usan 'override'
    public override void emitirsonido(){
        Console.WriteLine("meow meow");
    }

    public override void tecnicadecaza(){
        Console.WriteLine("el gato acecha y salta");
    }
    
    // implementacion de la interfaz idomestico
    public void jugarconhumano(){
        Console.WriteLine("el gato juega con una cuerda");
    }
}

File: classTypes4/java/src/Gato.java
public class Gato extends CarnivoroTerrestre implements Domestico {
    private String raza;
    
    // Constructor 1: inicializa con parámetros basicos
    public Gato(String nombreCientifico, double pesoKg, String raza) {
        super(nombreCientifico, pesoKg, 2); // Gestación de 2 meses
        this.raza = raza;
    }
    
    // constructor 2: sobrecarga, inicializa solo con el peso y usa valores por defecto para otros
    public Gato(double pesoKg) {
        this("felis catus", pesoKg, "Comun"); // Llama al Constructor 1
        System.out.println("sobrecarga usada: Gato comun creado con peso especifico.");
    }

    // metodos obligatorios implementados
    @Override
    public void emitirSonido() {
        System.out.println("Meow Meow");
    }

    @Override
    public void tecnicaDeCaza() {
        System.out.println("el gato acecha y salta");
    }
    
    @Override
    public void jugarConHumano() {
        System.out.println("el gato juega con una cuerda");
    }
}